Python/suggestions.c

```diff
@@ -181,9 +181,9 @@ offer_suggestions_for_name_error(PyNameErrorObject *exc) {
 PyObject *_Py_Offer_Suggestions(PyObject *exception) {
     PyObject *result = NULL;
     assert(!PyErr_Occurred());
-    if (PyErr_GivenExceptionMatches(exception, PyExc_AttributeError)) {
+    if (Py_TYPE(exception) == (PyTypeObject*)PyExc_AttributeError) {
         result = offer_suggestions_for_attribute_error((PyAttributeErrorObject *) exception);
-    } else if (PyErr_GivenExceptionMatches(exception, PyExc_NameError)) {
+    } else if (Py_TYPE(exception) == (PyTypeObject*)PyExc_NameError) {
```

Comment:
Maybe add a comment to explain why you don't want to handle UnboundLocalError here. I understand thta you should visit the globals and/or builtins to locate the proper name, and it's too complicated, right?

Reply:
No, is because with `UnboundLocalError` doesn't make sense to give suggestions: the variable is correctly typed, is just that is not defined yet. But in general I don;t want to handle subclasses here as is too complicated